### PR TITLE
Document coding style, add indent script, and Travis indent check, and apply coding style

### LINKED ISSUE #40: Choose, document, and use a style guide for provided code
Currently there is some inconsistency in formatting/style for the tests and sample solutions. We should choose a style guide, document it in README.md, and ensure that it is applied consistently on the track.

### PR DESCRIPTION
**Pull request by ryanplusplus**
Close #40

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,27 @@ Exercism problems in C
 
 ## Contributing Guide
 
-Please see the [contributing guide](https://github.com/exercism/x-api/blob/master/CONTRIBUTING.md#the-exercise-data)
+Please see the Exercism [contributing guide](https://github.com/exercism/x-api/blob/master/CONTRIBUTING.md#the-exercise-data) for general contribution tips.
+
+## Coding Style
```
Comment (ryanplusplus) [inline README.md]: Added overview of coding style

File: bin/verify-indent [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
```
Comment (ryanplusplus) [inline bin/verify-indent]: This is the script that Travis runs to ensure files are formatted properly

File: .travis.yml [hunk 3]
```diff
@@ -3,3 +3,4 @@ language: bash
 script:
 - bin/fetch-configlet
 - bin/configlet .
+- bin/verify-indent
```
Comment (ryanplusplus) [inline .travis.yml]: This configures Travis to run the indentation/style check

File: indent.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
```
Comment (ryanplusplus) [inline indent.sh]: This is the script you can run prior to committing to ensure proper style

File: README.md [hunk 5]
```diff
@@ -4,7 +4,27 @@ Exercism problems in C
 
 ## Contributing Guide
 
-Please see the [contributing guide](https://github.com/exercism/x-api/blob/master/CONTRIBUTING.md#the-exercise-data)
+Please see the Exercism [contributing guide](https://github.com/exercism/x-api/blob/master/CONTRIBUTING.md#the-exercise-data) for general contribution tips.
```
Comment (patricksjackson) [inline README.md]: I think this should point to the x-common contributing guide, https://github.com/exercism/x-common/blob/master/CONTRIBUTING.md

That seems to be where the most recent instructions are.

**Comment by patricksjackson** [comment]
This looks great :+1: I tested the scripts locally and everything seems fine.

The one thing we should do before merging is intentionally hit the failure case on Travis to make sure it works. Just push a slight change to the formatting of one of the *.c files and let the Travis build fail, we'll still be able to see the failure after it's taken out.

File: README.md (hunk 5 above)
Comment (ryanplusplus) [inline README.md]: Good catch. Fixed.

**Comment by ryanplusplus** [comment]
Sample Travis failure here:
https://travis-ci.org/exercism/xc/builds/157757693

File: exercises/leap/src/example.h [hunk 6]
```diff
@@ -4,4 +4,4 @@
 #include <stdbool.h>
 bool leap_year(int year);
 
-#endif // _LEAP_H
+#endif                          // _LEAP_H
```
Comment (palisaide) [inline exercises/leap/src/example.h]: Should we be nixing these comments on the include guards?

Comment (ryanplusplus) [inline exercises/leap/src/example.h]: Yes, I think so.

Comment (ryanplusplus) [inline exercises/leap/src/example.h]: Nixed

**Comment by palisaide** [comment]
Looks good 👍

**Comment by patricksjackson** [comment]
LGTM too. Squash all the commits and then we can merge it.

**Comment by ryanplusplus** [comment]
Actually, GitHub now lets you do a squash merge from the web interface: https://github.com/blog/2141-squash-your-commits. Just click on the right side of the merge button where the down arrow is and select "Squash and merge". No more manual squashing!

**Comment by patricksjackson** [comment]
Well that's certainly convenient :)

**Pull request merged**